src/redis-fast-driver.cc

```diff
@@ -276,17 +276,41 @@ NAN_METHOD(RedisConnector::RedisCmd) {
 		//LOG("str: \"%s\"\n", *str);
 		//LOG("len %u\n", len);
 		//LOG("bufused %zu\n", bufused);
-		if(bufused + len >= bufsize) {
+		if(bufused + len > bufsize) {
 			//increase buf size
 			// LOG("buf needed %zu\n", bufused + len);
 			// LOG("bufsize is not big enough, current: %zu ", bufsize);
 			// bufsize = bufsize * 2;
-			bufsize = ((bufused + len) / 256 + 1) * 256;
+			// bufsize = ((bufused + len) / 256 + 1) * 256;
+			if (i == arraylen - 1) {
+				// last element
+				// only (bufused + len) is really needed but give it a bit of headroom 
+				// since its likely a similar command will be fired again (and it might
+				// be slightly larger)
+				bufsize = 1.2 * (bufused + len);
```

Comment:
Nope, 1.2 multiplier is a bad strategy, for the best performance we should allocate memory aligned blocks. (Align is 64, so my 256 is aligned too, but in your case 4096*1.2 = 4915 which is not good)

Example from StackOverflow:
```
// cache line
#define ALIGN 64

void *aligned_malloc(int size) {
    void *mem = malloc(size+ALIGN+sizeof(void*));
    void **ptr = (void**)((uintptr_t)(mem+ALIGN+sizeof(void*)) & ~(ALIGN-1));
    ptr[-1] = mem;
    return ptr;
}

void aligned_free(void *ptr) {
    free(((void**)ptr)[-1]);
}
```